src/types/CodepointWidthDetector.cpp

```diff
@@ -4,6 +4,311 @@
 #include "precomp.h"
 #include "inc/CodepointWidthDetector.hpp"
 
+namespace
+{
+    // used to store range data in CodepointWidthDetector's internal map
+    struct UnicodeRange final
+    {
+        unsigned int lowerBound;
+        unsigned int upperBound;
+        CodepointWidth width;
+    };
+
+    static bool operator<(const UnicodeRange& range, const unsigned int searchTerm)
+    {
+        return range.upperBound < searchTerm;
+    }
+
+    static constexpr std::array<UnicodeRange, 285> s_wideAndAmbiguousTable{
+        // generated from http://www.unicode.org/Public/UCD/latest/ucd/EastAsianWidth.txt

```

Comment:
Probably add a note here that says that anything _not_ in this map is assumed to be narrow

Reply:
```suggestion
        // generated from http://www.unicode.org/Public/UCD/latest/ucd/EastAsianWidth.txt
        // anything not present here is presumed to be Narrow.
```